File: Hippo/Views/WebtoonDetailView.m
//
//  WebtoonDetailView.m
//  Hippo
//
//  Created by 전수열 on 13. 9. 6..
//  Copyright (c) 2013년 Joyfl. All rights reserved.
//

#import "WebtoonDetailView.h"

@implementation WebtoonDetailView

- (id)init
{
	self = [super initWithFrame:CGRectMake( 0, 0, 320, 60 )];
	self.backgroundColor = [UIColor colorWithHex:0xF6F6F6 alpha:0.8];
	
	self.thumbnailView = [[UIImageView alloc] initWithFrame:CGRectMake( 0, 0, 55, 60 )];
	[self addSubview:self.thumbnailView];
	
	self.titleLabel = [[UILabel alloc] initWithPosition:CGPointMake( 65, 4 ) maxWidth:190];
	[self addSubview:self.titleLabel];
	
	self.artistLabel = [[UILabel alloc] initWithPosition:CGPointMake( 65, 24 ) maxWidth:190];
	self.artistLabel.font = [UIFont systemFontOfSize:12];
	self.artistLabel.textColor = [UIColor darkGrayColor];
	[self addSubview:self.artistLabel];
	
	self.portalIconView = [[UIImageView alloc] initWithFrame:CGRectMake( 65, 41, 12, 12 )];
	[self addSubview:self.portalIconView];
	
	self.weekdayLabel = [[UILabel alloc] initWithPosition:CGPointMake( 81, 41 ) maxWidth:170];
	self.weekdayLabel.font = [UIFont systemFontOfSize:11];
	self.weekdayLabel.textColor = [UIColor darkGrayColor];
	[self addSubview:self.weekdayLabel];
	
	UIView *borderView = [[UIView alloc] initWithFrame:CGRectMake( 0, 59.5, 320, 0.5 )];
	borderView.backgroundColor = [UIColor colorWithHex:0xA7A7A7 alpha:0.9];
	[self addSubview:borderView];
	
	return self;
}

- (void)setWebtoon:(Webtoon *)webtoon
{
	_webtoon = webtoon;
	
	self.titleLabel.text = self.webtoon.title;
	[self.titleLabel sizeToFit];
	
	[self.thumbnailView setImageWithURL:[NSURL URLWithString:self.webtoon.thumbnail_url] placeholderImage:nil];
	
	self.titleLabel.text = self.webtoon.title;
	[self.titleLabel sizeToFit];
	
	if( [self.webtoon.artists count] )
	{
		NSMutableArray *artists = [NSMutableArray array];
		for( NSDictionary *artist in self.webtoon.artists ) {
			[artists addObject:[artist objectForKey:@"name"]];
		}
		self.artistLabel.text = [artists componentsJoinedByString:@", "];
		[self.artistLabel sizeToFit];
	}
	
	self.portalIconView.image = [UIImage imageNamed:[NSString stringWithFormat:@"icon_%@.png", self.webtoon.portal]];
	
	if( self.webtoon.finished.boolValue )
	{
		self.weekdayLabel.text = @"완결";
	}
	else
	{
		NSMutableArray *weekdays = [NSMutableArray array];
		for( NSInteger i = 1; i < 7; i++ )
		{
			NSString *weekday = HippoWeekdays[i];
			if( [[self.webtoon valueForKey:weekday] boolValue] ) {
				[weekdays addObject:NSLocalizedString( weekday.uppercaseString, nil )];
			}
		}
		NSLog( @"weekday : %@", [weekdays componentsJoinedByString:@", "] );
		self.weekdayLabel.text = [weekdays componentsJoinedByString:@", "];
	}
	[self.weekdayLabel sizeToFit];
}

@end

File: Hippo/Views/WebtoonDetailView.swift
//
//  WebtoonDetailView.swift
//  Hippo
//
//  Created by 전수열 on 10/5/14.
//  Copyright (c) 2014 Joyfl. All rights reserved.
//

import Foundation

class WebtoonDetailView: UIView {

    private var _webtoon: Webtoon?
    var webtoon: Webtoon? {
        get {
            return _webtoon?
        }
        set {
            _webtoon = newValue
            let placeholder = UIImage(color: UIColor.whiteColor())
            if newValue!.picture? == nil {
                self.thumbnailView.image = placeholder
            } else {
                let url = NSURL(string: newValue!.picture!.url)
                self.thumbnailView.setImageWithURL(url, placeholderImage: placeholder)
            }
            self.titleLabel.text = newValue!.title
            self.artistLabel.text = newValue!.artistText()
            self.portalIconView.image = UIImage(named: "icon_\(newValue!.portal)")
            self.weekdayLabel.text = newValue!.weekdayText()
        }
    }

    let thumbnailView = UIImageView()
    let titleLabel = UILabel()
    let artistLabel = UILabel()
    let portalIconView = UIImageView()
    let weekdayLabel = UILabel()
    let borderView = UIView()

    convenience override init() {
        self.init(frame: CGRectMake(0, 0, 320, 60))
    }

    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor(hex: 0xf6f6f6, alpha: 0.8)

        self.addSubview(self.thumbnailView)
        self.addSubview(self.titleLabel)
        self.addSubview(self.artistLabel)
        self.addSubview(self.portalIconView)
        self.addSubview(self.weekdayLabel)
        self.addSubview(self.borderView)

        self.thumbnailView.snp_makeConstraints { make in
            make.top.equalTo(0)
            make.size.equalTo(CGSizeMake(55, 60))
            return
        }

        self.titleLabel.snp_makeConstraints { make in
            make.left.equalTo(64)
            make.top.equalTo(4)
            make.width.lessThanOrEqualTo(190)
            return
        }

        self.artistLabel.font = UIFont.systemFontOfSize(12)
        self.artistLabel.textColor = UIColor.darkGrayColor()
        self.artistLabel.snp_makeConstraints { make in
            make.left.equalTo(64)
            make.top.equalTo(24)
            make.width.lessThanOrEqualTo(190)
            return
        }

        self.portalIconView.snp_makeConstraints { make in
            make.left.equalTo(64)
            make.top.equalTo(41)
            make.size.equalTo(CGSizeMake(12, 12))
            return
        }

        self.weekdayLabel.font = UIFont.systemFontOfSize(11)
        self.weekdayLabel.textColor = UIColor.darkGrayColor()
        self.weekdayLabel.snp_makeConstraints { make in
            make.left.equalTo(81)
            make.top.equalTo(41)
            make.width.lessThanOrEqualTo(170)
            return
        }

        self.borderView.backgroundColor = UIColor(hex: 0xa7a7a7, alpha: 0.9)
        self.borderView.snp_makeConstraints { make in
            make.width.equalTo(self)
            make.height.equalTo(0.5)
            make.bottom.equalTo(self)
            return
        }
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}